src/modules/claim/services/claim-issuance.service.ts

```diff
@@ -0,0 +1,150 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import jwt from 'jsonwebtoken';
+import { IClaimIssuance, DecodedClaimToken } from '../claim.types';
+import { RoleService } from '../../role/role.service';
+import { Logger } from '../../logger/logger.service';
+import { ClaimIssueDTO, NewClaimIssueDTO } from '../claim.dto';
+import { RoleClaim } from '../entities/roleClaim.entity';
+import { ClaimService } from './claim.service';
+import { ClaimHandleResult } from '../claim-handle-result.dto';
+
+@Injectable()
+export class ClaimIssuanceService {
+  constructor(
+    private readonly roleService: RoleService,
+    private readonly logger: Logger,
+    @InjectRepository(RoleClaim)
+    private readonly roleClaimRepository: Repository<RoleClaim>
+  ) {
+    this.logger.setContext(ClaimIssuanceService.name);
+  }
+
+  /**
+   * Handles claim issuance request saving and updates.
+   * Two scenarios are handled - issue requested claim and issue not-requested claim
+   * @param rq IClaimIssuance request
+   */
+  public async handleClaimIssuanceRequest(
```

Comment:
IMO there is not much need in this method. We can call one of the handles right after getting `previouslyRequestedClaim`